src/context/client.rs

```diff
@@ -0,0 +1,38 @@
+use crate::{
+    Context, RedisModule_GetClientCertificate, RedisModule_GetClientId,
+    RedisModule_GetClientInfoById, RedisModule_GetClientNameById,
+    RedisModule_GetClientUserNameById, ValkeyString,
+};
+use std::os::raw::c_void;
+
+impl Context {
```

Comment:
Wanted to add a note about `ValkeyModule_GetClientNameById`, `ValkeyModule_GetClientUserNameById` and the `ValkeyModule_GetClientCertificate` APIs.

These APIs use `autoMemoryAdd` on the ValkeyModuleString pointer in the core to release automatically when the callback returns.

So, after the callback (e.g. command, server event handler, unblock client event cb, etc), all of these ValkeyModuleString pointers will be freed

Reply:
so do you want to just add a comment to the code explaining this?

Reply:
Yes, if we are going with the current approach (where they are freed automatically after the callback), it would be helpful to mention that in a comment - I am good with this

I tried to look at other [examples](https://github.com/search?q=repo%3Avalkey-io%2Fvalkeymodule-rs+%22-%3E+ValkeyString%22&type=code) in this package, so we don't have a convention for this so far: 

https://github.com/valkey-io/valkeymodule-rs/blob/6f3a241d6b5be5b9f9b4ccd5e4399ce4214f37a5/src/digest.rs#L21 - We use retain underneath to hold onto the ValkeyModuleString.

https://github.com/valkey-io/valkeymodule-rs/blob/6f3a241d6b5be5b9f9b4ccd5e4399ce4214f37a5/src/context/mod.rs#L780 - We do not retain the ValkeyModuleString here

Reply:
good point, added the comment